chat-core/src/extractFrontMatter.ts

```diff
@@ -1,15 +1,15 @@
 import * as matter from "gray-matter";
 import toml from "toml";
 
-export function extractFrontMatter(
+export function extractFrontMatter<T = Record<string, unknown>>(
```

Comment:
I doubt we want this to be any T (number, string, ...) but rather any object?
```suggestion
export function extractFrontMatter<T extends Record<string, unknown> = Record<string, unknown>>(
```